storage/iterator.go

```diff
@@ -0,0 +1,18 @@
+package storage
+
+import "context"
+
+// Iterator represents a cursor for generic iteration over a sequence of items.
+type Iterator interface {
```

Comment:
is it beneficial to have these methods in their own interface?

Reply:
It's nice to hide the implementation of Batched vs Merging iterator from the storage functions calling them. It's a pretty stable interface, based on it not having changed in Cedar in forever.